### PR TITLE
Add operational hours

### LINKED ISSUE #403: Implement operational hours
1. Pick system's operational hours
2. Change mock to real value
3. Add min. operational hours to System's table
4. Make operational hours red if not compliant with system
5. Style name? Unsynchronized? updated?

### PR DESCRIPTION
**Pull request by LaChope**
Resolves #403 and #310, also related to #416

### CODE REVIEW

**Comment by LaChope** [comment]
@blcham just FYI, no need to review now, it is still WIP

File: src/utils/utils.tsx [hunk 1]
```diff
@@ -73,15 +73,12 @@ const getModifiedList = <T extends { name: string }>(items: T[], selected: strin
...
-  getModifiedList(systems, selected);
-
-export const getModifiedFaultTreesList = (faultTrees: FaultTree[], selected: string | null) => {
-  if(!faultTrees || !selected)
-    return faultTrees;
-  return faultTrees.filter(f => f?.system?.iri == selected);
-}
+export const getModifiedSystemsList = (systems: System[], selected: System) => getModifiedList(systems, selected);
 
+export const getModifiedFaultTreesList = (faultTrees: FaultTree[], selected: System) => {
```
Comment (blcham) [inline src/utils/utils.tsx:78]: We have getModifiedSystemsList and getModifiedFaultTreesList (CC @Kasmadei ): 
-- basically the same name without any comment on what it does, but they do fundamentally different things :(. This is confusing and can lead to errors easily.
-- getModifiedSystemsList -- reorders elements based on selected one
-- getModifiedFaultTreesList -- filter elements based on an attribute

File: src/components/dashboard/content/list/SystemOverview.tsx [hunk 2]
```diff
@@ -70,11 +70,7 @@ const SystemOverview = () => {
       </Box>
 
       {selectedView === "table" ? (
-        <FaultTreeAndSystemOverviewTable
-          selectedSystem={selectedSystem?.iri}
-          systems={systems}
-          handleSystemContextMenu={handleContextMenu}
-        />
+        <FaultTreeAndSystemOverviewTable systems={systems} handleSystemContextMenu={handleContextMenu} />
```
[the remark this answers is not in the document]
Comment (blcham) [reply src/components/dashboard/content/list/SystemOverview.tsx:73]: I will create ticket for that.

File: src/utils/utils.tsx (hunk 1 above)
Comment (blcham) [reply src/utils/utils.tsx:78]: I will create ticket for that.

**Review by blcham — CHANGES_REQUESTED** [review]
There is only one issue related to you code directly, other issues i will try to target as separate issues.

File: src/components/editor/faultTree/canvas/EditorCanvas.tsx [hunk 3]
```diff
@@ -249,6 +258,40 @@ const EditorCanvas = ({
...
+      }
+      return prevState;
+    });
+  }, [faultTree]);
+
+  const handleMinOperationalHoursChange = (event) => {
+    const newValue = event.target.value;
+    setUpdatedMinOperationalHours(newValue);
+    if (newValue !== faultTree?.operationalDataFilter?.minOperationalHours) {
+      setInputColor("red");
```
[the remark this answers is not in the document]
Comment (LaChope) [reply src/components/editor/faultTree/canvas/EditorCanvas.tsx]: @blcham I refactored adding `synchronized` and `notSynchronized` directly in the theme

File: public/locales/cs/translation.json [hunk 4]
```diff
@@ -84,7 +84,8 @@
     "lastEditor": "Poslední editor",
     "status": "Stav",
     "edit": "Upravit",
-    "created": "Vytvořeno"
+    "created": "Vytvořeno",
+    "operationalHours": "Provozní Doba"
```
Comment (LaChope) [inline public/locales/cs/translation.json:88]: @grotskat Can you revise this?

File: src/components/editor/faultTree/canvas/EditorCanvas.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (blcham) [reply src/components/editor/faultTree/canvas/EditorCanvas.tsx]: ok

File: src/utils/utils.tsx (hunk 1 above)
Comment (blcham) [reply src/utils/utils.tsx:79]: Will be resolved in https://github.com/kbss-cvut/fta-fmea-ui/pull/422

File: public/locales/cs/translation.json (hunk 4 above)
Comment (grotskat) [inline public/locales/cs/translation.json:88]:
```suggestion
    "operationalHours": "Doba provozu"
```

**Review by grotskat — DISMISSED** [review]

**Comment by blcham** [comment]
@LaChope does it make sense to you to improve the git history by squashing some of the commits to together

File: src/components/dashboard/content/list/SystemOverview.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (blcham) [reply src/components/dashboard/content/list/SystemOverview.tsx:73]: Will be resolved by https://github.com/kbss-cvut/fta-fmea-ui/issues/423

**Review by blcham — APPROVED** [review]
LGTM

@LaChope does it make sense to you to improve the git history by squashing some of the commits to together + do not forget to rebase

**Comment by LaChope** [comment]
> LGTM
> 
> @LaChope does it make sense to you to improve the git history by squashing some of the commits to together + do not forget to rebase

@blcham it would make sense to do it, but is it possible to squash when it was already pushed? Or I have to use --force, right?

**Comment by blcham** [comment]
@LaChope  you have to do force in this case

**Comment by LaChope** [comment]
@blcham I did squash but not so many commits in the end, I think history might be useful for later.

**Comment by blcham** [comment]
OK, although I thought you would squash the last commit.

**Comment by LaChope** [comment]
Will do next time, using interactive rebase rather than squash feature.

**Pull request merged**